exonum-java-binding/core/src/test/java/com/exonum/binding/core/runtime/TransactionMethodExtractorTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.runtime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.core.service.Node;
+import com.exonum.binding.core.service.Service;
+import com.exonum.binding.core.storage.database.Snapshot;
+import com.exonum.binding.core.transaction.TransactionContext;
+import com.exonum.binding.core.transaction.TransactionMethod;
+import io.vertx.ext.web.Router;
+import org.junit.jupiter.api.Test;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+class TransactionMethodExtractorTest {
+
+  @Test
+  void validServiceMethodExtraction() throws Exception {
+    Map<Integer, MethodHandle> transactions =
+        TransactionMethodExtractor.extractTransactionMethods(ValidService.class);
+    Method transactionMethod =
+        ValidService.class.getMethod("transactionMethod", byte[].class, TransactionContext.class);
+    MethodHandles.Lookup lookup = MethodHandles.lookup();
+    MethodHandle methodHandle = lookup.unreflect(transactionMethod);
+    assertThat(transactions).hasSize(1);
+    MethodHandle actualMethodHandle = transactions.get(ValidService.TRANSACTION_ID);
+    assertThat(actualMethodHandle.toString()).isEqualTo(methodHandle.toString());
```

Comment:
I had `assertThat(transactions).containsExactly(entry(ValidService.TRANSACTION_ID, methodHandle));` initially, but `methodHandles` are not equal to each other. Is there a better way to check this?

Reply:
If _invocation_ of the transaction method were implemented as the task specifies,
it could be tested by invoking the operation and verifying that it was indeed invoked:

```java
Object service = spy(new Service());
TransactionInvoker invoker = new TransactionInvoker(service);
invoker.invoke(id, context, bytes);

verify(service).operation(context, bytes);
```